tools/StaticAnalysis/Exceptions/Az.Monitor/BreakingChangeIssues.csv

```diff
@@ -0,0 +1,5 @@
+"AssemblyFileName","ClassName","Target","Severity","ProblemId","Description","Remediation"
+"Az.Monitor","Microsoft.Azure.Commands.Insights.Metrics.GetAzureRmMetricCommand","Get-AzMetric","0","3000","The type of property 'Unit' of type 'Microsoft.Azure.Commands.Insights.OutputClasses.PSMetric' has changed from 'Microsoft.Azure.Management.Monitor.Models.Unit' to 'System.String'.","Change the type of property 'Unit' back to 'Microsoft.Azure.Management.Monitor.Models.Unit'."
+"Az.Monitor","Microsoft.Azure.Commands.Insights.Metrics.GetAzureRmMetricDefinitionCommand","Get-AzMetricDefinition","0","3000","The type of property 'Unit' of type 'Microsoft.Azure.Commands.Insights.OutputClasses.PSMetricDefinition' has changed from 'System.Nullable`1[Microsoft.Azure.Management.Monitor.Models.Unit]' to 'System.String'.","Change the type of property 'Unit' back to 'System.Nullable`1[Microsoft.Azure.Management.Monitor.Models.Unit]'."
+"Az.Monitor","Microsoft.Azure.Commands.Insights.Alerts.NewAzureRmMetricAlertRuleV2CriteriaCommand","New-AzMetricAlertRuleV2Criteria","0","3000","The type of property 'TimeAggregation' of type 'Microsoft.Azure.Commands.Insights.OutputClasses.PSMetricCriteria' has changed from 'System.Object' to 'System.String'.","Change the type of property 'TimeAggregation' back to 'System.Object'."
+"Az.Monitor","Microsoft.Azure.Commands.Insights.Alerts.NewAzureRmMetricAlertRuleV2CriteriaCommand","New-AzMetricAlertRuleV2Criteria","0","3000","The type of property 'TimeAggregation' of type 'Microsoft.Azure.Commands.Insights.OutputClasses.PSDynamicMetricCriteria' has changed from 'System.Object' to 'System.String'.","Change the type of property 'TimeAggregation' back to 'System.Object'."
```

Comment:
Next release is a major release so we allow breaking changes to be checked in, but we also need to provide the user with a migration document.
Please fill in a table like this:
|Cmdlet|Breaking Change | Old Usage | New Usage |
|-|-|-|-|
|Get-AzMetric|The type of property 'Unit' of type 'Microsoft.Azure.Commands.Insights.OutputClasses.PSMetric' has changed from 'Microsoft.Azure.Management.Monitor.Models.Unit' to 'System.String'|{powershell script that demonstrates the old usage}| {powershell script that demonstrates the new usage}|

Reply:
<html>
<body>
<!--StartFragment-->

Cmdlet | Breaking Change | Old Usage | New Usage
-- | -- | -- | --
Get-AzMetric | The type of property 'Unit' of type 'Microsoft.Azure.Commands.Insights.OutputClasses.PSMetric' has changed from 'Microsoft.Azure.Management.Monitor.Models.Unit' to 'System.String' | {powershell script that demonstrates the old usage} | {powershell script that demonstrates the new usage}
Get-AzMetric | The type of property 'Unit' of type 'Microsoft.Azure.Commands.Insights.OutputClasses.PSMetricDefinition' has changed from 'System.Nullable`1[Microsoft.Azure.Management.Monitor.Models.Unit]' to 'System.String'| [Cmdlet("Get", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "MetricDefinition"), OutputType(typeof(PSMetricDefinition))] | [Cmdlet("Get", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "MetricDefinition"), OutputType(typeof(PSMetricDefinition))]
Get-AzMetric | The type of property 'TimeAggregation' of type 'Microsoft.Azure.Commands.Insights.OutputClasses.PSMetricCriteria' has changed from 'System.Object' to 'System.String' | [Cmdlet("New", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "MetricAlertRuleV2Criteria", DefaultParameterSetName = StaticThresholdParameterSet) | [Cmdlet("New", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "MetricAlertRuleV2Criteria", DefaultParameterSetName = StaticThresholdParameterSet)
Get-AzMetric | The type of property 'TimeAggregation' of type 'Microsoft.Azure.Commands.Insights.OutputClasses.PSDynamicMetricCriteria' has changed from 'System.Object' to 'System.String'| {powershell script that demonstrates the old usage} | {powershell script that demonstrates the new usage}

<!--EndFragment-->
</body>
</html>